### ISSUE TITLE
UnusedVariable when definition is after use

### ISSUE THREAD

**Issue by cverbiest** [Riverside-Software/sonar-openedge#510]
Following sample code shows 2 false positive for the UnusedVariable  rule

Unused variable AllowAddNew	test_unused_var.cls
Unused variable SavedAllowAddNew	test_unused_var.cls

The properties are used before their definition which is valid with the this-object reference

```
block-level on error undo, throw.

class be.cce.SmartTools.SmartComponents.test_unused_var: 

    define public property ReadOnly as logical no-undo
        get.
        set(iReadOnly as logical):

            if iReadOnly
            then do:
                this-object:SavedAllowAddNew = this-object:AllowAddNew.
                this-object:AllowAddNew = no.
            end.
            else do:
                if this-object:SavedAllowAddNew ne ?
                then this-object:AllowAddNew = this-object:SavedAllowAddNew.
            end.
            this-object:ReadOnly = iReadOnly.

        end set.

    define private property SavedAllowAddNew as logical no-undo initial ?
        get.
        set.

    define private property AllowAddNew as logical no-undo
        get.
        set.


end class.
```

**Comment by gquerret** [comment]
Reproduced

**Comment by gquerret** [comment]
Fixed
